src/briefcase/commands/base.py

```diff
@@ -192,7 +194,7 @@ def __init__(
         # These are abstracted to enable testing without patching.
         self.cookiecutter = cookiecutter
         self.requests = requests
-        self.input = Console(enabled=input_enabled)
+        self.input = console or Console(enabled=input_enabled)
```

Comment:
Is there any value in having this fallback? Given that the purpose of this PR is to ensure we only ever have a single Console and Logger, preserving a mechanism in the constructor to create a new Console seems like a potential footgun. Should `logger` and `console` be promoted to required arguments (and, I guess, input_enabled removed as an argument)?

Reply:
If they became required arguments, wouldn't that require dozens of updates to all the places where commands are created in the unit tests?

I think a cleaner solution to this is for the Log and the Console to become attributes of a single "services" or "tools" object rather than the Command, as discussed at https://github.com/beeware/briefcase/pull/801#pullrequestreview-1096865165. 
But this PR is still a good improvement in itself: the bigger picture can be left to #801.

Reply:
The impact on tests is definitely a concern; however, I'd prefer mildly more inconvenient tests over potentially dangerous public APIs.

I'm not sure I see how moving Log and Console to ToolsCache makes this substantially better though - we'd still need to modify every Command instantiation; the only difference would be providing a single tools/services object, rather than a Log and a Console.

Reply:
ngl - part of my thinking was minimizing impact to tests....but that isn't a great reason by any means.

Skimming the use of command attributes from within tools, the use of `input` and `logger` are common. So, if `input` and `logger` are not moved in to the new `ToolCache`, they would need to be part of the initialization. Given that, it may be better to move everything over to reference `input` and `logger` from `command.tools`...

Reply:
That's a really good point - I hadn't considered that. I guess that seals the case for input and logger being on the tools object.

The downside is that we'll need to use `self.tools.logger.info()` everywhere that we want to log inside a command... which isn't great API. I guess we could make `self.log_info()` a member method on BaseCommand?

Reply:
I don't think there's any need for that, the Command constructor can just set `self.logger = self.tools.logger`, and all the existing logging code can remain unchanged.

Reply:
In light of this discussion, I am:
 - moving these changes back to #801
 - removing footgun defaults for `logger` and `console`